### PR TITLE
Updates to sanitization job

### PR DESCRIPTION
**Pull request by quiiver**
- Removes blank queries
- Tracks blank queries in metadata
- Updates deps to recent versions
- Run tests in python 3.10

### CODE REVIEW

File: nightly-job/query_sanitization.py [hunk 1]
```diff
@@ -163,6 +165,33 @@ def stream_search_terms():
...
+UNSANITIZED_QUERY_STATS = """
+SELECT
+    COUNT(DISTINCT search_logs.jsonPayload.fields.rid) AS total_term_count,
+    COUNTIF(TRIM(search_logs.jsonPayload.fields.query) = "") AS total_blank_count,
+FROM `suggest-searches-prod-a30f.logs.stdout` AS search_logs
+WHERE 
+    jsonPayload.type = "web.suggest.request"
+    --Specifically get the previous day's data
+    AND timestamp >= DATE_ADD(CURRENT_TIMESTAMP(), INTERVAL -1 DAY)
+    AND DATE(timestamp) < CURRENT_DATE()
```
Comment (dzeber) [inline nightly-job/query_sanitization.py:178]: We should change the date filter to `DATE(timestamp) = DATE_ADD(CURRENT_DATE(), INTERVAL -1 DAY)` here and above.

This is a separate issue I noticed, where the current date filter is dependent on the job start time. I mentioned this in [DENG-3279](https://mozilla-hub.atlassian.net/browse/DENG-3279) but I should have opened a separate ticket.

**Review by dzeber — APPROVED** [review]
Looks good!

It would be good to update the docstrings in `stream_search_terms` and `record_job_metadata` as well to mention handling of blank queries.

**Comment by chelseatroy** [comment]
Pulling down to run now!

**Comment by chelseatroy** [comment]
OK, had to sort out some bigquery client stuff, back on this and running

File: nightly-job/query_sanitization.py (hunk 1 above)
Comment (chelseatroy) [reply nightly-job/query_sanitization.py:178]: It _is_, and if I recall correctly we did it like this for a specific reason, but for the life of me now I can't recall the reason 😅

**Comment by chelseatroy** [comment]
Did some additional cleanup and updates. Now simulating the full flow to make sure everything works 🤩

File: nightly-job/query_sanitization.py (hunk 1 above)
Comment (dzeber) [reply nightly-job/query_sanitization.py:179]: Ok yeah, if this is necessary we should discuss or document - it does come at the cost of silently dropping all search terms submitted betweeen midnight and the job start time.

Comment (quiiver) [reply nightly-job/query_sanitization.py]: Updated the date logic in my most recent commit.

**Comment by quiiver** [comment]
Ran the job locally and it completed without error

**Comment by chelseatroy** [comment]
@quiiver I updated the schema of the downstream table (PR here: https://github.com/mozilla/bigquery-etl/pull/5367)

I think that one has to merge before we merge this one to avoid error in prod, but I can't think of any other changes to adjacent repos that have to happen.

**Comment by quiiver** [comment]
@chelseatroy i already have an open pr that you approved 😁

https://github.com/mozilla/bigquery-etl/pull/5337

and here is the associated change control process ticket that needs approval from mark before that can be landed. 

https://mozilla-hub.atlassian.net/browse/DSRE-1594

**Comment by chelseatroy** [comment]
@quiiver AH! I THOUGHT this changeset looked familiar!! 😂 

I'll delete my PR!

**Pull request merged**